### PR TITLE
Add CMake generated files for the Xcode build system to .gitignore

### PR DESCRIPTION
**Pull request by JadenGeller**
@bdash

### CODE REVIEW

**Comment by bdash** [comment]
I think you can simplify this to:
```
CMakeScripts/
realm-object-store.xcodeproj/
*.build/
Debug/
```

You'll also want to add:
```
Release/
RelWithDebInfo/
MinSizeRel/
```

It'd be worth splitting the CMake generated files and Xcode build output into the appropriate section of the .gitignore.

**Comment by JadenGeller** [comment]
Awesome feedback @bdash! Thank you! I made the suggested changes.

File: .gitignore [hunk 1]
```diff
@@ -7,8 +7,14 @@ Makefile
 build.ninja
 cmake_install.cmake
 rules.ninja
+CMakeScripts/
+realm-object-store.xcodeproj/
 
 # Build products
 src/librealm-object-store.dylib
 src/librealm-object-store-static.a
-tests/tests
```
Comment (bdash) [inline .gitignore]: Looks like you lost this in the process.

Comment (JadenGeller) [inline .gitignore]: Oh, I should've waited longer before merging...

Comment (JadenGeller) [inline .gitignore]: Fixed https://github.com/realm/realm-object-store/pull/123 😳

**Pull request merged**